### PR TITLE
update cmake build docs

### PR DESCRIPTION
**Pull request by Surendrajat**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
I've updated cutter building with cmake documentation as per the new dir structure and backend change.

**Test plan (required)**
- Build cutter with cmake
- Build from VS code build/debug option after `Cmake: Configure`

**Closing issues**

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

**Review by xarkes — APPROVED** [review]

**Review by xarkes — CHANGES_REQUESTED** [review]
I followed the instructions
```
$ git clone --recurse-submodules https://github.com/rizinorg/cutter
$ cd cutter
$ mkdir build && cd build
$ cmake -DCUTTER_USE_BUNDLED_RIZIN=ON ../src
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:1 (include):
  include could not find load file:

    QMakeConfigureFile
```

**Comment by Surendrajat** [comment]
@xarkes you should use `cmake -DCUTTER_USE_BUNDLED_RIZIN=ON ..`

**Review by ITAYC0HEN — APPROVED** [review]

**Comment by xarkes** [comment]
Yes I am just very bad at reading diffs.

**Review by xarkes — APPROVED** [review]
lgtm

File: docs/source/building.rst [hunk 1]
```diff
@@ -77,18 +77,18 @@ The recommended way to build Cutter on Linux is by using CMake. Simply invoke CM
...
-   cmake -DCUTTER_USE_BUNDLED_RADARE2=ON ../src
+   cmake -DCUTTER_USE_BUNDLED_RIZIN=ON ..
    cmake --build .
 
 If your operating system has a newer version of CMake (> v3.12) you can use this cleaner solution:
 
 .. code:: sh
 
-   cmake -S src -B build -DCUTTER_USE_BUNDLED_RADARE2=ON
+   cmake -S . -B build -DCUTTER_USE_BUNDLED_RIZIN=ON
```
Comment (karliss) [inline docs/source/building.rst:87]: `-S .` is redundant.

Comment (Surendrajat) [reply docs/source/building.rst]: removed.

**Review by karliss — APPROVED** [review]

**Pull request merged**